OPHD/States/MapViewStateDraw.cpp

```diff
@@ -172,11 +172,12 @@ void MapViewState::drawResourceInfo()
 	}
 
 	// Capacity (Storage, Food, Energy)
+	const auto& sm = NAS2D::Utility<StructureManager>::get();
```

Comment:
Not sure how other people feel, though I generally prefer variable names to be spelled out in full. For the definition, the abbreviation may be reasonable clear, but at the point of use that context is lost.

Reply:
This one is iffy for me -- I don't generally object but the scope here is so limited it almost seems like overkill to spell it out. The intention of the variable in the first place was to 1) avoid multiple lookups and 2) help shorten the verbosity of the code. I'm not sure if fully spelling it out here would help more than it would hinder considering the limited scope?

Reply:
True, the scope is quite limited here, though refactoring does tend to move variables around, so scope can sometimes expand. It's nice to be able to expand scope without having to rename things at the same time. Besides, if you use auto complete, it doesn't make much of a typing difference. And if you don't use autocomplete, the speed of programming is never limited by the speed of typing. More likely, the speed of understanding already written code will be more relevant. In that case, seeing the use "sm.someMethod" will be less immediately clear than "structureManager.someMethod". The definition is mercifully close, but it still requires cross referencing the use with the defining line.